### PR TITLE
Recognize sd card initialization on Repetier fw

### PR DESCRIPTION
**Pull request by arekm**
Repetier firmware uses different text for sd card initialization:

Send: N10 M21*33
Recv: ok 10
Recv: Card successfully initialized.

Recognize that in octoprint, so sd card operations become available in
UI.

Tested with Repetier firmware 1.0.4.

  * [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [ ] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's `devel` branch if it's a completely
    new feature, or `maintenance` if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against `master` or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of `master`, `maintenance`, or `devel`
    please), e.g. `dev/my_new_feature` or `fix/my_bugfix`
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [ ] If your changes include style sheets: You have modified the
    `.less` source files, not the `.css` files (those are generated
    with `lessc`)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [ ] You have run the existing unit tests against your changes and
    nothing broke
  * [ ] You have added yourself to the `AUTHORS.md` file :)

### CODE REVIEW

**Review by foosel — CHANGES_REQUESTED** [review]
As things are right now, if Repetier firmware is detected and automatic firmware detection is enabled, OctoPrint will always assume that an SD is available, so this PR would turn into a no-op. 

If we go for native support for Repetier SD card messaging and handling, that part of the autodetection needs to be removed (see lines 2787 and following in `comm.py`), and matching of error messages also needs to be added. As far as I can say looking at the firmware source, the relevant strings for init failure and such are [here](https://github.com/repetier/Repetier-Firmware/blob/7cb3741323ba45537eceb4d54e99b1cbdedf2739/src/ArduinoAVR/Repetier/SDCard.cpp#L93-L112) and for umounting [here](https://github.com/repetier/Repetier-Firmware/blob/7cb3741323ba45537eceb4d54e99b1cbdedf2739/src/ArduinoAVR/Repetier/SDCard.cpp#L146).

**Comment by arekm** [comment]
In my case plugin does M21 command which ends up with "Recv: SD init fail". In result octoprint sees that and disables sd card. 

Then another M21 gets "Card successfully initialized" and that's not recognized by octoprint, so sd card support stays disabled.

But I can also do what you suggest and go fully for "native support".

**Comment by foosel** [comment]
Ah, yeah, you are right, I didn't spot [this line](https://github.com/repetier/Repetier-Firmware/blob/7cb3741323ba45537eceb4d54e99b1cbdedf2739/src/ArduinoAVR/Repetier/SDCard.cpp#L91) where that is generated and which always gets generated for any of the error cases. So that should indeed be covered. In that case I think we only the `sdAlwaysAvailable` handling removed in the auto detection routine. Detecting an unmount would be nice-to-have though ([see generated message in this line](https://github.com/repetier/Repetier-Firmware/blob/7cb3741323ba45537eceb4d54e99b1cbdedf2739/src/ArduinoAVR/Repetier/SDCard.cpp#L146)) ;)

**Comment by foosel** [comment]
Also, please make sure to install `pre-commit`, your changes currently fail some checks.

**Comment by arekm** [comment]
Suggestions implemented.

(no idea why github check still says "1 failing"; pre-commit run --hook-stage manual --all-files passes here locally)

**Comment by foosel** [comment]
That would be the E2E tests that are sadly turning more and more flaky on CI and are driving me slowly insane. Not your fault, big todo on my own task list :)

File: src/octoprint/util/comm.py [hunk 1]
```diff
@@ -2928,11 +2923,15 @@ def convert_line(line):
                         "SD init fail" in line
                         or "volume.init failed" in line
                         or "openRoot failed" in line
+                        or "SD Card unmounted" in line
+                        or "SD card released" in line
```
Comment (foosel) [inline src/octoprint/util/comm.py:2927]: Where did you find that string? I'm looking at the Marlin source but so far haven't found anything similar in there (doesn't even seem to have any string for umounting sadly).

Comment (arekm) [reply src/octoprint/util/comm.py:2927]: [here](https://github.com/prusa3d/Prusa-Firmware/blob/MK3/Firmware/cardreader.cpp#L267)

Comment (foosel) [reply src/octoprint/util/comm.py:2927]: Ah, Prusa Firmware, gotcha, thanks!

**Review by foosel — APPROVED** [review]

**Pull request merged**